internal/workflow/types.go

```diff
@@ -0,0 +1,217 @@
+	ExecutionStartToCloseTimeout time.Duration
+
+	// TaskStartToCloseTimeout - The decision task timeout for the child workflow.
+	// Optional: default is 10s if this is not provided (or if 0 is provided).
+	TaskStartToCloseTimeout time.Duration
+
+	// WaitForCancellation - Whether to wait for cancelled child workflow to be ended (child workflow can be ended
+	// as: completed/failed/timedout/terminated/canceled)
+	// Optional: default false
+	WaitForCancellation bool
+
+	// WorkflowIDReusePolicy - Whether server allow reuse of workflow ID, can be useful
+	// for dedup logic if set to WorkflowIdReusePolicyRejectDuplicate
+	WorkflowIDReusePolicy int
+
+	// RetryPolicy specify how to retry child workflow if error happens.
+	// Optional: default is no retry
+	RetryPolicy *RetryPolicy
+
+	// CronSchedule - Optional cron schedule for workflow. If a cron schedule is specified, the workflow will run
+	// as a cron based on the schedule. The scheduling will be based on UTC time. Schedule for next run only happen
+	// after the current run is completed/failed/timeout. If a RetryPolicy is also supplied, and the workflow failed
+	// or timeout, the workflow will be retried based on the retry policy. While the workflow is retrying, it won't
+	// schedule its next run. If next schedule is due while workflow is running (or retrying), then it will skip that
+	// schedule. Cron workflow will not stop until it is terminated or cancelled (by returning cadence.CanceledError).
+	// The cron spec is as following:
+	// ┌───────────── minute (0 - 59)
+	// │ ┌───────────── hour (0 - 23)
+	// │ │ ┌───────────── day of the month (1 - 31)
+	// │ │ │ ┌───────────── month (1 - 12)
+	// │ │ │ │ ┌───────────── day of the week (0 - 6) (Sunday to Saturday)
+	// │ │ │ │ │
+	// │ │ │ │ │
+	// * * * * *
+	CronSchedule string
+
+	// Memo - Optional non-indexed info that will be shown in list workflow.
+	Memo map[string]interface{}
+
+	// SearchAttributes - Optional indexed info that can be used in query of List/Scan/Count workflow APIs (only
+	// supported when Cadence server is using ElasticSearch). The key and value type must be registered on Cadence server side.
+	// Use GetSearchAttributes API to get valid key and corresponding value type.
+	SearchAttributes map[string]interface{}
+
+	// ParentClosePolicy - Optional policy to decide what to do for the child.
+	// Default is Terminate (if onboarded to this feature)
+	ParentClosePolicy int
+
+	// Bugports allows opt-in enabling of older, possibly buggy behavior, primarily intended to allow temporarily
+	// emulating old behavior until a fix is deployed.
+	//
+	// Bugports are always deprecated and may be removed in future versions.
+	// Generally speaking they will *likely* remain in place for one minor version, and then they may be removed to
+	// allow cleaning up the additional code complexity that they cause.
+	//
+	// Deprecated: All bugports are always deprecated and may be removed at any time.
+	Bugports Bugports
```

Comment:
nit: you can omit deprecated fields. also the fields that are not likely to be leveraged by starlark-worker users to keep it lean.